core/CronArchive.php

```diff
@@ -938,6 +934,77 @@ public function isThereExistingValidPeriod(Parameters $params)
         return !empty($idArchive);
     }
 
+    // public for tests
+    public function repairInvalidationsIfNeeded($archiveToProcess)
+    {
+        $table = Common::prefixTable('archive_invalidations');
+
+        $bind = [
+            $archiveToProcess['idsite'],
+            $archiveToProcess['name'],
+            $archiveToProcess['period'],
+            $archiveToProcess['date1'],
+            $archiveToProcess['date2'],
+        ];
+
+        $reportClause = '';
+        if (!empty($archiveToProcess['report'])) {
+            $reportClause = " AND report = ?";
+            $bind[] = $archiveToProcess['report'];
+        }
+
+        $sql = "SELECT DISTINCT period FROM `$table` WHERE idsite = ? AND name = ? AND period > ? AND ? >= date1 AND date2 >= ? $reportClause";
+
+        $higherPeriods = Db::fetchAll($sql, $bind);
+        $higherPeriods = array_column($higherPeriods, 'period');
+        $higherPeriods = array_flip($higherPeriods);
```

Comment:
not important but could have done instead below an `in_array($id, $higherPeriods)` . not too important though

Reply:
sure, thought this was a bit faster, but for an array w/ 3 items at most, probably not